### ISSUE TITLE
notifyPropertyChanged on devices as no effect in the gateway

### ISSUE THREAD

**Issue by jarod68** [mozilla-iot/gateway#975]
Hi,

I am trying to implement an adapter using a set of devices accessible through a REST API with SSE events.

The setValue of my Property class is working well, returning a promise and calling notifyPropertyChanged.

However I am using the module 'eventsource' to attach listeners on SSE events when a device property changed on the API side. Thus I only need to update the mozilla devices properties. In these listeners I call onProp.setCachedValue(newValue); and device.notifyPropertyChanged(onProp); the value is correctly set but I need to refresh the webinterface to have te update.

Is that a bug in the gateway or I am miss-understanding something in my adapter ?

I use revision 43fa84e of the gateway.

Thanks,

**Comment by sogaani** [comment]
@jarod68  Hi,
The Gateway dose not implement `EventSource` APIs. However, we can use [`websocket` APIs](https://iot.mozilla.org/wot/#web-thing-websocket-api) to get SSE.

**Comment by sogaani** [comment]
@jarod68
Could you share minimal codes to reproduce your problem?

**Comment by jarod68** [comment]
HI @sogaani ,
Thanks for replying me, finally the bug was that I used integers as things ids. On notify, things_controller.js (line 251) was comparing integer vs strings and test failed despite that the id where the same as an integer point of view.

```
  function onPropertyChanged(property) {
    if (property.device.id !== thingId) {
      return;
    }...
```
I changed my ids allocation with a string prefix and now everything is fine. However I think that preventing this kind of "javascript stuff" the "Device" class should check the id at instanciation.

See u,
